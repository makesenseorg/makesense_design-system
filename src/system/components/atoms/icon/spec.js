import { shallowMount } from "@vue/test-utils";
import Comp from "./Icon.vue";

describe("Icon.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: 'clock'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("displays the icon", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: 'clock'
      }
    });

    expect(wrapper.element.innerHTML).not.toBeUndefined();
  });

  it("uses shortcut for old font awesome names", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: 'arrowDown'
      }
    });
    expect(wrapper.classes()).toContain("icon-arrow-down");
  });

  it("sets the color class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: 'clock',
        color: "primary",
      },
    });
    expect(wrapper.classes()).toContain("icon--color-primary");
  });

  it("throws on invalid color", () => {
    const color = Comp.props.color;
    expect(color.validator).toBeInstanceOf(Function);
    expect(color.validator("primary")).toBeTruthy();
    expect(color.validator("red")).toBeFalsy();
  });

  it("sets the size", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: 'clock',
        size: "50",
      },
    });

    expect(wrapper.attributes('width')).toBe("50");
    expect(wrapper.attributes('height')).toBe("50");
    
  });

  it("sets the stroke width", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: 'clock',
        strokeWidth: "5",
      },
    });

    expect(wrapper.attributes('stroke-width')).toBe("5");
    
  });

  
});