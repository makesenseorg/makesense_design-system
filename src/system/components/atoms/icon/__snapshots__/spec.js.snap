// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.vue matches snapshot 1`] = `
<svg
  class="icon icon-clock icon--color-inherit"
  fill="none"
  height="24"
  role="img"
  stroke-linecap="round"
  stroke-linejoin="round"
  stroke-width="2"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="clock"
  />
  <circle
    cx="12"
    cy="12"
    r="10"
  />
  <polyline
    points="12 6 12 12 16 14"
  />
</svg>
`;
